src/Integration.Vsix/CFamily/FileConfig.cs

```diff
@@ -30,19 +30,36 @@ partial class CFamilyHelper
     {
         internal class FileConfig
         {
-            public static FileConfig TryGet(ProjectItem dteProjectItem, string absoluteFilePath)
+            public static FileConfig TryGet(ILogger logger, ProjectItem dteProjectItem, string absoluteFilePath)
             {
                 var vcProject = dteProjectItem.ContainingProject.Object as VCProject;
                 var file = dteProjectItem.Object as VCFile;
                 if (vcProject == null || file == null)
                 {
                     return null;
                 }
-
                 var vcConfig = vcProject.ActiveConfiguration;
+                var projectKind = vcConfig.ConfigurationType;
+                // Unknown projects represent all the unspported project type like makefile projects
+                if (projectKind == ConfigurationTypes.typeUnknown)
+                {
+                    logger.WriteLine(@"Project's ""Configuration type"" is not supported.");
+                    return null;
+                }
+                // "ClCompile" for source files and "ClCompile" for header files
```

Comment:
Is that a mistake in the comment? did you mean to say "ClInclude"?

Reply:
yes it is